packages/driver/src/dom/visibility.ts

```diff
@@ -10,19 +9,74 @@ const { isElement, isBody, isHTML, isOption, isOptgroup, getParent, getFirstPare
 const fixedOrAbsoluteRe = /(fixed|absolute)/
 
 const OVERFLOW_PROPS = ['hidden', 'scroll', 'auto']
+const { wrap } = $jquery
+let optionsObject = {
+  checkOpacity: true,
+  checkVisibilityCSS: true,
+  opacityPropert: true,
+  contentVisibilityAuto: true,
+}
 
 const isVisible = (el) => {
-  return !isHidden(el, 'isVisible()')
+  ensureEl(el, 'isVisible()')
+
+  const optionIsVisible = checkIsOptionVisible(el)
+
+  if (optionIsVisible === true) {
+    return true
+  }
+
+  if (optionIsVisible > 1) {
+    return false
+  }
+
+  if (!(typeof el.checkVisibility === 'function')) {
+    return !isHidden(el, 'isVisible()')
+  }
+
+  if (isStrictlyNotVisibile(el)) {
+    return false
+  }
+
+  if (typeof el.checkVisibility === 'function') {
+    return el.checkVisibility(optionsObject) && !isNotVisibleBecauseOfAncestors(el)
+  }
+
+  return isHiddenByAncestors(el, 'isVisible', optionsObject)
 }
 
-const { wrap } = $jquery
+const checkIsOptionVisible = (el) => {
+  // an option is considered visible if its parent select is visible
+  if (isOption(el) || isOptgroup(el)) {
+    const $el = $jquery.wrap(el)
+
+    if (elHasDisplayNone($el)) {
+      return 2
+    }
+
+    // if its parent select is visible, then it's not hidden
+    const $select = getFirstParentWithTagName($el, 'select')
+
+    if ($select && $select.length) {
+      // if the select is hidden, the options in it are not visible too
+      if (isStrictlyHidden($select)) {
+        return 2 //this signal not visible
+      }
+    }
+
+    return true //this signal visible
+  }
+
+  return 0 //this signal not option element
+}
```

Comment:
This seems like it could be simplified. The return of several numbers for each instance doesn't seem so simple. Can we make it true/false check? With the other checks in another function?

Reply:
done